egui/src/widgets/slider.rs

```diff
@@ -255,6 +273,12 @@ impl<'a> Slider<'a> {
         if let Some(max_decimals) = self.max_decimals {
             value = emath::round_to_decimals(value, max_decimals);
         }
+        if let Some(step) = self.step {
+            let remainer = value % step;
+            if remainer != 0.0 {
+                value -= remainer;
+            };
```

Comment:
I believe this will have better numerical properties, and will also round better when clicking:

```suggestion
            value = (value / step).round() * step;
```